### PR TITLE
Add news and happyhour push notification to web_push

### PR DESCRIPTION
**Pull request by emedd33**
Remove the web_push model to include:

- Manage.py command deletezombiedevice for deleting inactive devices
- Move devices into customprofile for a M2M field
- Add news and happyhour subscription for push notification when created

### CODE REVIEW

File: chemie/customprofile/views.py [hunk 1]
```diff
@@ -111,7 +112,31 @@ def edit_profile(request):
...
+        if form.is_valid():
+            coffee_checked = form.cleaned_data['coffee_subscription']
+            news_checked = form.cleaned_data['news_subscription']
+            happyhour_checked = form.cleaned_data['happyhour_subscription']
+            
+            request.user.profile.coffee_subscription = coffee_checked
+            request.user.profile.news_subscription = news_checked
+            request.user.profile.happyhour_subscription = happyhour_checked
+            
+            request.user.profile.save()
```
Comment (sindreoy) [inline chemie/customprofile/views.py]: Replace 120-128 by form.save()

File: chemie/web_push/models.py [hunk 2]
```diff
@@ -21,18 +21,36 @@ def fifteen_minutes_has_passed(cls):
 
         # Checks if fifteen minutes have passed since last CoffeSubmission object was created
         fifteen_minutes = datetime.timedelta(0, 900)
-        last_coffee_submission_date = (datetime.datetime.utcnow() - fifteen_minutes).replace(tzinfo=pytz.UTC)
+        now = datetime.datetime.now()
+        last_coffee_submission_date = (now - fifteen_minutes).replace(tzinfo=pytz.timezone('Europe/Oslo'))
```
Comment (sindreoy) [inline chemie/web_push/models.py]: In ```settings/base.py```, we have a variable set for this:
```TIME_ZONE = "Europe/Oslo"```

File: chemie/web_push/models.py [hunk 3]
```diff
@@ -99,6 +109,61 @@ def is_active(self):
...
+        #         device = cls.objects.create(
+        #             apns_device=apns_device, 
+        #             owner=user,
+        #             gcm_device=None       
+        #         )
+        #         user.profile.devices.add(device)
+        #         status = 201
+
+        if cls.objects.filter(owner=user).count() > 5:
+            cls.objects.latest("-date_created").delete()
```
Comment (sindreoy) [inline chemie/web_push/models.py]: Seems like either the user has 0 devices and gets his device added, or he has > 5 devices, and his earliest device is deleted? What happens when 0 < count < 5 ? And why is his device removed without adding another?

File: chemie/customprofile/models.py [hunk 4]
```diff
@@ -137,6 +137,20 @@ class Profile(models.Model):
...
+        default=True, verbose_name="Aboner på kaffe pushvarsler"
+    )
+    
+    news_subscription = models.BooleanField(
+        default=True, verbose_name="Aboner på nyheter pushvarsler"
+    )
+
+    happyhour_subscription = models.BooleanField(
+        default=True, verbose_name="Aboner på happy hour pushvarsler"
+    )
```
Comment (sindreoy) [inline chemie/customprofile/models.py]: Inflexible to more types of subscription and has a bloating effect on the Profile model. Consider replacing by a ManyToManyField

File: chemie/web_push/models.py (hunk 2 above)
Comment (emedd33) [reply chemie/web_push/models.py]: I know, men fikk opp feilmelding hvis jeg ikke satt inn timezone. siden datetime.now() ikke setter timzone by default

File: chemie/web_push/models.py (hunk 3 above)
Comment (emedd33) [reply chemie/web_push/models.py]: Den kjører etter deivce added, så den legger til en device til brukeren også etterpå sjekker om brukern har flere enn 5 devicer, hvis han har det slettes den eldste av dem.  så denne koden kjører ikke hvis han har 0 til 5 devicer.

**Pull request merged**